### ISSUE TITLE
clientSecret required?

### ISSUE THREAD

**Issue by allComputableThings** [vishnuagbly/google_sign_in_all_platforms#12]
I'm a little confused about what this library is for. 
I'd like a library that performs login for all app clients (including Linux, Windows, MacOS, web and mobile), but the example demands a clientSecret which I assume shouldn't be distributed with the app, right?

https://github.com/vishnuagbly/google_sign_in_all_platforms/blob/main/google_sign_in_all_platforms/lib/google_sign_in_all_platforms.dart
```
  GoogleSignIn({GoogleSignInParams params = const GoogleSignInParams()})
      : assert(
          !isDesktop ||
              (params.clientSecret != null && params.clientId != null),
          'For Desktop, clientSecret and clientId cannot be null',
        ),
        assert(
          !Platform.isAndroid || params.clientId != null,
          'For Android, clientId cannot be null',
        ) 
```

**Comment by vishnuagbly** [comment]
Client secret is always provided with all apps, even with google_sign_in

**Comment by allComputableThings** [comment]
Not for webapps:
```
<!DOCTYPE html>
<html>
  <body>
    <script src="https://accounts.google.com/gsi/client" async defer onload="console.log('TODO: add onload function')"></script>
    <div id="g_id_onload"
         data-client_id="%(client_id)s"
         data-callback="handleCredentialResponse">
    </div>
    <div class="g_id_signin" data-type="standard"></div>
  </body>
  <div>Hi 😎</div>
</html>
```
Are mobile apps different?

**Comment by vishnuagbly** [comment]
Yeah if you look at the firebase implementation as well, it generates a
creds file, consisting of all the client creds

**Comment by allComputableThings** [comment]
To be clear, if I distribute an app that includes the client secret, I've made a huge mistake, right?
The "firebase implementation" that requires a client_secret is only for a dart backend service right? That should never be seen in frontend UI.

**Comment by allComputableThings** [comment]
Or to put it another way, if I was to produce a distributable Linux app, can I use this library for google login without a client_secret in the UI?

**Comment by vishnuagbly** [comment]
Currently I have not implemented it for linux..., Have not been able to
work on it for some time.., please feel free to contribute 🙏🏻

**Comment by allComputableThings** [comment]
I guess, I'd first like to understand what you think the inputs and outputs should be for all of the client login APIs. 
I'm pretty sure client_id is sufficient and thanks that client_secret in the UI is a security bug.

**Comment by tnc1997** [comment]
Just to provide more information, desktop applications require a client secret, but it may be included.

Alternatively you may proxy the token request via a confidential back-end that adds the client secret.

> Secrets that are statically included as part of an app distributed to multiple users should not be treated as confidential secrets, as one user may inspect their copy and learn the shared secret.  For this reason, and those stated in [Section 5.3.1 of [RFC6819]](https://datatracker.ietf.org/doc/html/rfc6819#section-5.3.1), it is NOT RECOMMENDED for authorization servers to require client authentication of public native apps clients using a shared secret, as this serves little value beyond client identification which is already provided by the "client_id" request parameter.

https://datatracker.ietf.org/doc/html/rfc8252#section-8.5

> Hey Matt,
> 
> We don't expect those secrets to stay secret—so far we're including them mostly so it's convenient to use with libraries today, and expect to stop requiring them at some point in the future.
> 
> Andrew

https://groups.google.com/g/oauth2-dev/c/HnFJJOvMfmA

> Installed apps are distributed to individual devices, and it is assumed that these apps cannot keep secrets. They can access Google APIs while the user is present at the app or when the app is running in the background. Incremental authorization with installed apps is not supported due to the fact that the client cannot keep the client_secret confidential.

https://github.com/tnc1997/flutter-google-sign-in-desktop
